### PR TITLE
prefix relative build path to enable builds from other dirs

### PR DESCRIPTION
**Pull request by timsurber**
I prefixed all relative build paths with the current project directory to enable building in QtCreator, or more generally in other directories.

Using QtCreator enables the use of automated checks for migrating to Qt6.

These are my goals in the future which are enabled by this PR:
- fix clang-tidy / clazy warnings
- migrate to Qt6
- move from more or less deprecated qmake to cmake
- include external libs via cmake
- update external libs

### CODE REVIEW

File: ain_imager_src/ain_imager.pro [hunk 1]
```diff
@@ -2,8 +2,8 @@ QT += core gui widgets network printsupport concurrent multimedia
 CONFIG += c++11 release app_bundle
 
 unix:mac {
-	CONFIG += app_bundle
```
Comment (rumengb) [inline ain_imager_src/ain_imager.pro:5]: In Ain, Indigo and Control panel we use TABs not 4 spaces. Please use tabs.

File: ain_imager_src/ain_imager.pro [hunk 2]
```diff
@@ -117,64 +116,70 @@ else: unix:!android: target.path = /usr/bin
...
-	../common_src/xisf.h \
-	../common_src/xml.h \
-	../common_src/pixelformat.h \
-	../external/qcustomplot/qcustomplot.h \
-	../common_src/coordconv.h \
-	../common_src/stretcher.h \
-	../common_src/image_stats.h \
-	../common_src/dslr_raw.h
-
-INCLUDEPATH += "../indigo/indigo_libs" + "../external" + "../external/libraw/" + "../external/lz4/" + "../common_src" + "../object_data" + "../ain_imager_src"
```
Comment (rumengb) [inline ain_imager_src/ain_imager.pro:153]: Why did you remove quotes? Any particular reason for that?

File: ain_viewer_src/ain_viewer.pro [hunk 3]
```diff
@@ -66,51 +65,56 @@ QML_DESIGNER_IMPORT_PATH =
...
+               $$PWD/../external/lz4/ \
+               $$PWD/../common_src \
+               $$PWD/../ain_viewer_src
+
+LIBS += -L$$PWD/../external/libraw/lib -L$$PWD/../../external/libraw/lib -L$$PWD/../../external/lz4 -L$$PWD/../external/lz4 -lraw -lz
 
 unix:!mac {
-	INCLUDEPATH += "../external/libjpeg"
-	LIBS += -L"../external/libjpeg/.libs" -L"../indigo/build/lib" -l:libindigo.a -lz -ljpeg -l:liblz4.a
-	#LIBS += -L"../external/libjpeg/.libs" -L"../indigo/build/lib" -lindigo -ljpeg -l:liblz4.a
```
Comment (rumengb) [inline ain_viewer_src/ain_viewer.pro:100]: I kept this for a reason. would you please put it back?

**Review by rumengb — CHANGES_REQUESTED** [review]

**Comment by rumengb** [comment]
Please note that you can not update external libs so easily.  Qcustom plot is stripped down and heavily customized as I do not need the bulk of it and I need Ain to be light weight. Also, libraw is a stripped-down version because I do not need 80% of its functionality and I am trying to keep Ain small and lightweight.  Libjpeg is held as a dependency because Qt native jpleg support crashes on some platforms (I do not remember which but I think it was Linux on Amd64 architecture) when converting jpeg to the internal format Ain needs.

**Comment by timsurber** [comment]
> Please note that you can not update external libs so easily. Qcustom plot is stripped down and heavily customized as I do not need the bulk of it and I need Ain to be light weight. Also, libraw is a stripped-down version because I do not need 80% of its functionality and I am trying to keep Ain small and lightweight. Libjpeg is held as a dependency because Qt native jpleg support crashes on some platforms (I do not remember which but I think it was Linux on Amd64 architecture) when converting jpeg to the internal format Ain needs.

Okay, good to know. I will look into it, when we cross that bridge.
One thing to note is that qcustomplot is GPL-licensed which does not allow relicensing under another license (like the INDIGO license).

**Comment by rumengb** [comment]
Sorry, Ain does not compile anymore on Linux and windows. 
In file included from ../../ain_viewer_src/main.cpp:25:
../../ain_viewer_src/viewerwindow.h:27:10: fatal error: indigo/indigo_bus.h: No such file or directory
 #include <indigo/indigo_bus.h>
          ^~~~~~~~~~~~~~~~~~~~~

I have fixed it, but please pay more attention to the PRs you make :)

**Comment by rumengb** [comment]
> Okay, good to know. I will look into it, when we cross that bridge. One thing to note is that qcustomplot is GPL-licensed which does not allow relicensing under another license (like the INDIGO license).

qcutomplot is not re-licensed it is just placed in the Ain source tree with the modifications. The file headers still contain the original author and the license and the source is available no GPL violation.

**Comment by timsurber** [comment]
> I have fixed it, but please pay more attention to the PRs you make :)

Thank you. I will try, starting in a new project always has difficulties.

**Comment by timsurber** [comment]
> > Okay, good to know. I will look into it, when we cross that bridge. One thing to note is that qcustomplot is GPL-licensed which does not allow relicensing under another license (like the INDIGO license).
> 
> qcutomplot is not re-licensed it is just placed in the Ain source tree with the modifications. The file headers still contain the original author and the license and the source is available no GPL violation.

I'm not sure this is correct. As I understand it, as soon as you link to one GPL library, the whole project has to be GPL.
This is not the case for LGPL

https://www.gnu.org/licenses/gpl-faq.en.html#IfLibraryIsGPL

**Comment by rumengb** [comment]
> I'm not sure this is correct. As I understand it, as soon as you link to one GPL library, the whole project has to be GPL. This is not the case for LGPL
> 
> https://www.gnu.org/licenses/gpl-faq.en.html#IfLibraryIsGPL

You are correct! I forgot GPL is like a cancer it spreads everywhere. I will rewrite qcustomplot at some point. it is ugly anyway :)

**Comment by timsurber** [comment]
Yeah GPL can be quite difficult.
QT6 includes a new Graph Module https://doc-snapshots.qt.io/qt6-dev/qtgraphs-2d-hellographs-example.html

I have no personal experience with it, but it looks quite nice.
